openml/config.py

```diff
@@ -2,23 +2,44 @@
 Store module level information like the API key, cache directory and the server
 """
 import logging
+import logging.handlers
 import os
+from typing import cast
 
 from io import StringIO
 import configparser
 from urllib.parse import urlparse
 
-
 logger = logging.getLogger(__name__)
-logging.basicConfig(
-    format='[%(levelname)s] [%(asctime)s:%(name)s] %('
-           'message)s', datefmt='%H:%M:%S')
+
+
+def configure_logging(console_output_level: int, file_output_level: int):
+    """ Sets the OpenML logger to DEBUG, with attached Stream- and FileHandler. """
+    openml_logger = logging.getLogger('openml')
+    openml_logger.setLevel(logging.DEBUG)
+    message_format = '[%(levelname)s] [%(asctime)s:%(name)s] %(message)s'
+    output_formatter = logging.Formatter(message_format, datefmt='%H:%M:%S')
+
+    console_stream = logging.StreamHandler()
+    console_stream.setFormatter(output_formatter)
+    console_stream.setLevel(console_output_level)
+
+    one_mb = 2**20
+    log_path = os.path.join(cache_directory, 'openml_python.log')
+    file_stream = logging.handlers.RotatingFileHandler(log_path, maxBytes=one_mb, backupCount=1)
+    file_stream.setLevel(file_output_level)
+    file_stream.setFormatter(output_formatter)
+
+    openml_logger.addHandler(console_stream)
+    openml_logger.addHandler(file_stream)
+
 
 # Default values!
 _defaults = {
     'apikey': None,
     'server': "https://www.openml.org/api/v1/xml",
-    'verbosity': 0,
+    'verbosity': logging.WARNING,
```

Comment:
Not sure if (and how) we should follow the standards we once set up for this: https://github.com/openml/OpenML/wiki/Client-API-Standards

Reply:
I was unaware of this document. I propose to map the verbosity for console logging (naturally) as:
 - 0, normal: `logging.WARNING`
 - 1, info: `logging.INFO`
 - 2 debug: `logging.DEBUG`

I suppose it makes sense to do this conversion in Python only, so that a user that uses both R and Python can share its configuration file across platforms (as was intended). It's only a minor inconvenience at our part to map the numbers.

The file specifies a *minimal* standard. So to me that implicitly reads that it may be expanded, and I think adding the `file_verbosity` is fine (but for consistency, probably also should follow 0/1/2 log levels).

Reply:
I agree.